app/src/main/java/com/github/se/signify/ui/screens/profile/SettingsScreen.kt

```diff
@@ -360,20 +364,55 @@ fun ConfirmationDialog(
  */
 @Composable
 fun LanguageSwitch(isFrench: Boolean, onLanguageChange: (Boolean) -> Unit) {
-  Row(
-      modifier =
-          Modifier.fillMaxWidth()
-              .padding(16.dp)
-              .clickable { onLanguageChange(!isFrench) }
-              .testTag("LanguageSwitch"),
-      verticalAlignment = Alignment.CenterVertically,
-      horizontalArrangement = Arrangement.SpaceBetween) {
-        Text(
-            text = if (isFrench) "FR" else "EN",
-            style = MaterialTheme.typography.bodyLarge,
-            color = MaterialTheme.colorScheme.onBackground)
-        Switch(checked = isFrench, onCheckedChange = { onLanguageChange(it) })
-      }
+  var expanded by remember { mutableStateOf(false) }
+  var selectedLanguage by remember { mutableStateOf(if (isFrench) "FR" else "EN") }
+
+  Column {
+    Row(
+        modifier =
+            Modifier.testTag("LanguageRow")
+                .fillMaxWidth()
+                .clickable { expanded = !expanded }
+                .padding(16.dp),
+        horizontalArrangement = Arrangement.SpaceBetween) {
+          Text(
+              text = selectedLanguage,
+              style = MaterialTheme.typography.bodyLarge,
+              color = MaterialTheme.colorScheme.onBackground)
+
+          Icon(
+              imageVector =
+                  if (expanded) Icons.Filled.KeyboardArrowDown
+                  else Icons.AutoMirrored.Filled.KeyboardArrowRight,
+              contentDescription = "Expand language menu",
```

Comment:
This string can be replaced with a string ressource.

Reply:
done in expand_language_menu_text putted in strings.xml